src/eduid/satosa/scimapi/statsd.py

```diff
@@ -0,0 +1,56 @@
+import logging
+import re
+from typing import Any, Mapping
+
+import satosa.internal
+from satosa.micro_services.base import ResponseMicroService
+
+from eduid.common.config.base import StatsConfigMixin
+from eduid.common.stats import init_app_stats
+
+logger = logging.getLogger(__name__)
+
+
+class requester_counter(ResponseMicroService):
+    """
+        A class to count the requesting SP.
+
+        Example configuration:
+    ```yaml
+        module: statsd.requester_counter
+        name: requester_counter
+        config:
+                app_name: idpproxy
+                stats_port: 8125
+                stats_host': monitor-fre-1.eduid.se
+
+    ```
+    """
+
+    def __init__(self, config: Mapping[str, Any], *args: Any, **kwargs: Any):
+
+        super().__init__(*args, **kwargs)
+
+        statsd_config = StatsConfigMixin(**config)
+        self.stats = init_app_stats(statsd_config)
+
+    def process(
+        self, context: satosa.context.Context, data: satosa.internal.InternalData
+    ) -> satosa.internal.InternalData:
+        try:
+            requester_entity_id = context.state.state_dict["SATOSA_BASE"]["requester"]
+        except KeyError:
+            logger.warn("Unable to determine the entityID for the SP requester")
+            return super().process(context, data)
+
+        # Graphite is picky about the characters in it's key names
+        requester_entity_id = re.sub(r"[^a-zA-Z0-9]", "_", requester_entity_id)
+        # For easier readability - only allow one underscore in a row
+        requester_entity_id = re.sub(r"_{2,}", "_", requester_entity_id)
```

Comment:
I am not of the use-case, but changing the identifier of the service will make it hard to understand which is the actual service in play. But at least you can see part of the identifier directly.

Would it be better to encode the identifier to base64? Doing that the characters will all be printable ASCII chars and the identifier is preserved intact. But to actually see the service identifier you need to decode it.

Reply:
I think that human readable identifiers are preferred but it's up to @johanlundberg, @sunetzacharias and pax to decide.

Know that this might cause collisions which I also think is OK for this context.

Reply:
Yeah, I think we need a human readable key as we want to graph it in Grafana directly from influxdb.